support-frontend/assets/pages/subscriptions-redemption/api.js

```diff
@@ -53,7 +53,11 @@ function doValidation(userCode: string, dispatch: Dispatch<Action>) {
 
 function validateUserCode(userCode: string, dispatch: Dispatch<Action>) {
   dispatch({ type: 'SET_USER_CODE', userCode });
-  doValidation(userCode, dispatch);
+  if (userCode.length === 13) {
```

Comment:
just a thought - I know why you are wary about adding knowledge about the code structure to the codebase.
since it's only that we don't want to hit zuora itself, and we already have partial validation here: https://github.com/guardian/support-frontend/blob/c1f856d7b573077b4fb2ff340b7beb7dec0cf7d5/support-models/src/main/scala/com/gu/support/redemptions/RedemptionCode.scala#L9
so would it make sense add the new "13" limit in the same place and do the check on the backend?
Then at least all the assumptions about the code structure are centralised.

Reply:
What do you think about having it in both places? I think there is value in not making unnecessary http requests so it's worth having it in the client but maybe we should have it in the server as well?

Reply:
good idea, both places would make sense at runtime even if not in the codebase twice so something like this in the template
```js
window.guardian.codeLength = ${RedemptionCode.length}
```

Reply:
this is just a general case of duplication of validation between client and server though

Reply:
It serves different purposes though, on the client it is preventing unnecessary http requests and on the server it is preventing us hitting Zuora unnecessarily if the client validation has failed for some reason.

Reply:
yeah that's what I realised in the end - I think we both agree it's sensible and appropraite to validate in both places, but we don't want to define the rules in two+ places in the source code due to maintainability and discoverabilty reasons.
Change looks good though 👍